.github/workflows/nf-core-linting.yml

```
@@ -0,0 +1,71 @@
+name: nf-core linting
+# This workflow is triggered on pushes and PRs to the repository.
+# It runs the `nf-core lint` tests to ensure that the module code meets the nf-core guidelines
+on: [push, pull_request]
+
+jobs:
+  changes:
+    name: Check for changes
+    runs-on: ubuntu-latest
+    outputs:
+      # Expose matched filters as job 'modules' output variable
+      modules: ${{ steps.filter.outputs.changes }}
+    steps:
+      - uses: actions/checkout@v2
+      - uses: dorny/paths-filter@v2
+        id: filter
+        with:
+          filters: 'tests/config/pytest_software.yml'
+
+  test:
+    runs-on: ubuntu-20.04
+    name: ${{ matrix.tags }} ${{ matrix.nxf_version }}
+    needs: changes
+    if: needs.changes.outputs.modules != '[]'
+    strategy:
+      fail-fast: false
+      matrix:
+        nxf_version: ['21.03.0-edge']
+        tags: ['${{ fromJson(needs.changes.outputs.modules) }}']
+    env:
+      NXF_ANSI_LOG: false
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python
```

Whitespace between the steps helps readability 👌🏻 
```suggestion

      - name: Set up Python
```